src/Symfony/Component/DependencyInjection/ParameterBag/ParameterBag.php

```diff
@@ -68,7 +66,7 @@ public function all()
      */
     public function get($name)
     {
-        $name = $this->normalizeName($name);
+        $name = (string) $name;
```

Comment:
Why these string casts?

Reply:
Previously implied by `strtolower`. And we actually rely on that, tests fail otherwise.

https://github.com/symfony/symfony/blob/master/src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php#L1558 leads to `has/getParameter($paramInstance)` call.

Reply:
then the dumper should be updated on the lowest branch for making it cast to string beforehand IMHO

Reply:
why? we don't use strict types (fortunately), so casting is just fine IMHO - and is functionally similar to adding the type hint on the signature (if we could)

Reply:
It's a matter of contract. If `@param string` means, we also support objects that can be cast to string, then we need to add the string cast everywhere in symfony. So I agree with @chalasr that this is a bug of the dumper.

Reply:
this is PHP.
If we agree that `@param string $foo` should be similar to `funtion(string $foo)`, then this cast is just required.
not doing so is a BC break already.

Reply:
Agree; without strict types it's the same as `fn(string $foo)`. Thus justified to do.

Reply:
I'm just saying string and castable object instance have not been treated the same everywhere.
For example there is no string cast in https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Routing/Route.php#L515
And there are thousands more of such code snippets. And it's basically an implementation detail if using an object works or not, nothing you can really rely on as an API user.

Reply:
Thats why I'm in favor of adding more type declarations as it makes thing at least consistent.

Reply:
I agree with @Tobion here.

Reply:
So do I, type declaration are nice. :)

Reply:
> I'm just saying string and castable object instance have not been treated the same everywhere.

The thing is we used `strtolower` before, which happily casts. Thus making it a supported feature, probably unintended but at this point we have to consider it.

Therefor i dont think it's part of this PR, which is now merged anyway. But maybe for the sake of consistency, we could cast beforehand, where needed. And deprecate passing stringish values or just ignore the fact it was possible before, regarding end users.

Reply:
In the past we usually rejected explicit type casts and let it be the user's responsibility to respect our API.

Reply:
And in the past also we had to revert as bugfix because we broke things with this policy...

Reply:
But we break BC in this case :) Technically at least, passing a stringish object actually worked as expected before. Core uses it :)

So yes there's an inconsistency between `$a[strtolower($k)]` and `$a[$k]` by design. I believe @nicolas-grekas comment "This is PHP" reflects that.

Now if everywhere we'd do `fn(string $k)` the consistency is gone, while preserving the stringish behavior. Id prefer doing that instead of removing it here first.

Reply:
Lets just keep it. This will always be a case-by-case decision until we have type declarations.